### DISCUSSION TITLE
Asking Support for more printers models

### DISCUSSION THREAD

**Discussion by EfraDiGmr** [Ircama/epson_print_conf#24] [Q&A]
HI, I'VE SEEN A LOT OF PEOPLE TALKING ABOUT YOUR WORK AND SAYING THAT IT WORKS LIKE A CHARM!!!.

I would like to bring my printer back to life too, but its models is not supported by the app (GUI?). So I like to gently ask support for the model L120, if its possible.

Thank you so much for you work and effort.

**Comment by CleverCoder** [comment]
Hmm... yeah, this project is a bit of a gem. I wish Epson published the codes and details to make this easier for people. Anyway, I'm busy tonight, otherwise I'd try to import the settings, but it looks like the "interesting" database with printer data here seems to contain data for the L120: https://codeberg.org/attachments/147f41a3-a6ea-45f6-8c2a-25bac4495a1d
There's instructions in the README on how it can be used to add new printer support, although it assumes programming experience.
(Is there a local copy of this data file in this project? It might be good to grab a copy in case it vanishes)
So, might be a matter of importing the printer details and creating a new configuration to support it.

**Reply by EfraDiGmr** [reply]
Oh!, I would really appreciate some help with the coding since I'm kinda a newby in python (only knowledge in maths things).

Anyway, thanks for the help and the resources. If I eventualy get my printer working again I'm sharing my results here.

Thanks for the time.

**Comment by Ircama** [comment]
This repository is focused on Python code development, I have no time to provide usage support. It includes data converter tools based on reusable code to generate formats compatible with the data schema designed here, but with no responsibility for any source data. Anyway, since the code accesses the printer EEPROM, it is essential that data are correct.

Generally, data contributions will be merged only if specifically verifiable and I am against massive imports.

**Comment by Ted7474** [comment]
Please add EPSON L3111

**Reply by Ircama** [reply]
Follow [How to import an external printer configuration DB](https://github.com/Ircama/epson_print_conf?tab=readme-ov-file#how-to-import-an-external-printer-configuration-db), importing the reported XML configuration file.

Check whether the printer is managed in such a way and, in case of errors, paste the related application report. 

Then press "Printer status" in the UI. The status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

Press "Detect Access Keys". After the operation is completed, the status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

Press "Detect configuration". After the operation is completed, the status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

If you have notes or logs from any other tools you tested or additional hints, please add them.

**Comment by killeriq** [comment]
possible to add L3250 ?

Or does it help if the "Detect Access Keys" output (takes like 30m+ ) will be provided to you?

Thanks!

**Reply by killeriq** [reply]
![image](https://github.com/user-attachments/assets/6046e894-7f48-427f-920d-71c6d4057a03)

**Reply by Ircama** [reply]
[There is no known way](https://github.com/Ircama/epson_print_conf?tab=readme-ov-file#known-incompatible-models) at the moment to manage the L3250 model.

**Comment by programuser-01** [comment]
Please add support for the xp-800 model Epson Printer

**Reply by Ircama** [reply]
I do not have that model and I need information to understand whether it can be managed, I can't figure it out blindly without any data.

Follow [How to import an external printer configuration DB](https://github.com/Ircama/epson_print_conf?tab=readme-ov-file#how-to-import-an-external-printer-configuration-db), importing the reported XML configuration file.

Check whether the printer is managed in such a way and, in case of errors, paste the related application report. 

Then press "Printer status" in the UI. The status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

Press "Detect Access Keys". After the operation is completed, the status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

Press "Detect configuration". After the operation is completed, the status box will show a tree view with printer data. Select the status box, right button of the mouse, then copy all data and paste them here.

If you have notes or logs from any other tools you tested or additional hints, please add them.

**Comment by pilok27** [comment]
Please can you add the ET2820 onto the list?
Thanks a lot for your sharing and working

**Reply by Ircama** [reply]
On the ET2820 model I have the same requests [already asked](https://github.com/Ircama/epson_print_conf/discussions/24#discussioncomment-11513882). If the "Detect Access Keys" operation fails with the "Could not detect read_key" message, I need to know which are the right sequences to read that EEPROM, considering that at the moment the program is based on [the specifications documented in the README](https://github.com/Ircama/epson_print_conf?tab=readme-ov-file#byte-sequences).

**Comment by yagwit** [comment]
I'm not a CP. How do I donate to this most worthy cause? I have epson 8700,4800,3760, 2800.  Only the 8700 has been spared from Epson's planned obsolescence. I knew about the issue long ago. I thought I would just keep the tanks empty, and sparkly clean. It seems it's not really for the tank. The printer is coded to fail at a certain point. 
I hope this project picks up steam. Thank you so much for making a start here.

**Reply by Ircama** [reply]
> I'm not a CP. How do I donate to this most worthy cause? I have epson 8700,4800,3760, 2800. Only the 8700 has been spared from Epson's planned obsolescence. I knew about the issue long ago. I thought I would just keep the tanks empty, and sparkly clean. It seems it's not really for the tank. The printer is coded to fail at a certain point. I hope this project picks up steam. Thank you so much for making a start here.

The only acceptable contributions are:

- coding pull requests
- [output of the discovery buttons of the application](https://github.com/Ircama/epson_print_conf/discussions/24#discussioncomment-11513882) for printers that do not work
- hints about the right sequences to read the EEPROM, considering that currently the program is based on [the specifications documented in the README](https://github.com/Ircama/epson_print_conf?tab=readme-ov-file#byte-sequences).

There is an additional test that can be possibly done: at the moment, before writing data to the EEPROM, I attempt to read data from it in order to verify the EEPROM access and log useful information to be able to reverse the operation; for some models, this read operation fails (unknown reason). Anyway, it is possible to write directly to the EEPROM without the preliminary reading operation. For instance, to reset the waste ink levels, do the following:

- open the app
- load the appropriate printer configuration file (from the menu)
- select the correct printer model
- select the correct IP address
- from the menu, Settings > Show printer parameters of the selected model (F2); check that *raw_waste_reset* is available
- press "Printer status" to verify a valid output from the printer
- from the menu, Settings > Debug Shell (F7)
- at the prompt, enter the following (single line): `for oid, value in self.printer.parm["raw_waste_reset"].items(): self.printer.snmp_mib(self.printer.eeprom_oid_write_address(oid, value, label="raw_waste_reset"), label="raw_waste_reset")`
- paste the output here

**Comment by Shiinuz** [comment]


**Reply by Ircama** [reply]
Your PC might be configured to use the old Internet Explorer 11 as the default browser. I suggest changing it to a newer application.

**Reply by Shiinuz** [reply]
Just seen that i posted in the wrong place. I opened an "issue" and posted my workaround, just in case anyone encounters the same.
